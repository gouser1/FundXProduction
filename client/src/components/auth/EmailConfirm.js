import React from 'react';
import { withRouter } from 'react-router-dom';
import {
  Grid,
  Box,
  Button,
  Paper,
  withStyles,
  IconButton,
} from '@material-ui/core';
import LogoNav from '../../images/home/LogoNav.png';
import useStyles from './authStyle';

const EmailConfirm = (props) => {
  const { history } = props;
  const handleLinkClick = (pageURL) => {
    history.push(pageURL);
  };

  const classes = useStyles();

  const LogoButton = withStyles(() => ({}))(IconButton);

  return (
    <div className={classes.root}>
      <Grid
        container
        spacing={0}
        direction='column'
        alignItems='center'
        justifyContent='center'
        style={{ minHeight: '100vh' }}
      >
        <LogoButton
          onClick={() => handleLinkClick('/')}
          style={{ backgroundColor: 'transparent' }}
        >
          <img src={LogoNav} alt='' width='130px' height='35px' />
        </LogoButton>
        <Paper elevation={10} className={classes.paper4}>
          <h1 className={classes.h1}>Welcome to FundX!</h1>
          <Box pb={1}>
            <h2 className={classes.h2}>
              Please check your inbox for a confirmation email. Click the link
              in the email to confirm your email address.
            </h2>{' '}
          </Box>
          <h2 className={classes.h2}>After you confirm click Continue.</h2>
          <Grid item xs={12}>
            <Box textAlign='center' pt={4}>
              <Button
                onClick={() => handleLinkClick('/')}
                className={classes.button}
              >
                Continue
              </Button>
              <Button
                onClick={() => handleLinkClick('/')}
                className={classes.button2}
                variant='outlined'
              >
                Re-send confirmation email
              </Button>
            </Box>
          </Grid>
        </Paper>
      </Grid>
    </div>
  );
};

export default withRouter(EmailConfirm);
