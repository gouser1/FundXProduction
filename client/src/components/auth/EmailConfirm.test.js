import React from 'react';
import renderer from 'react-test-renderer';

import EmailConfirm from './EmailConfirm';

it('renders correctly', () => {
  const tree = renderer.create(<EmailConfirm />).toJSON();
  expect(tree).toMatchSnapshot();
});
