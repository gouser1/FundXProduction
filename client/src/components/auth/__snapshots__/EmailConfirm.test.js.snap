// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt=""
          height="35px"
          src="LogoNav.png"
          width="130px"
        />
      </span>
    </button>
    <div
      className="MuiPaper-root makeStyles-paper4-10 MuiPaper-elevation10 MuiPaper-rounded"
    >
      <h1
        className="makeStyles-h1-5"
      >
        Welcome to FundX!
      </h1>
      <div
        className="MuiBox-root MuiBox-root-13"
      >
        <h2
          className="makeStyles-h2-6"
        >
          Please check your inbox for a confirmation email. Click the link in the email to confirm your email address.
        </h2>
         
      </div>
      <h2
        className="makeStyles-h2-6"
      >
        After you confirm click Continue.
      </h2>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiBox-root MuiBox-root-14"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Continue
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button2-4"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Re-send confirmation email
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
